invenio_rdm_records/services/config.py

```diff
@@ -255,6 +234,214 @@ class RDMSearchVersionsOptions(SearchVersionsOptions, SearchOptionsMixin):
     ] + SearchVersionsOptions.params_interpreters_cls
 
 
+#
+# Helper link definitions
+#
+
+
+class RecordPIDLink(Link):
+    """Record external PID link."""
+
+    def vars(self, record, vars):
+        """Add record PID to vars."""
+        vars.update(
+            {
+                f"pid_{scheme}": pid["identifier"]
+                for (scheme, pid) in record.pids.items()
+            }
+        )
+        vars.update(
+            {
+                f"parent_pid_{scheme}": pid["identifier"]
+                for (scheme, pid) in record.parent.pids.items()
+            }
+        )
+
+
+class ThumbnailLinks:
+    """RDM thumbnail links dictionary.
+
+    Adopts the interface of a Link but not one.
+    """
+
+    link_for_thumbnail = EndpointLink(
+        "iiif.image_api",
+        params=["uuid", "region", "size", "rotation", "quality", "image_format"],
+    )
+
+    def __init__(self, sizes=None, when=None):
+        """Constructor."""
+        self._sizes = sizes
+        self._when_func = when
+
+    def should_render(self, obj, context):
+        """Determine if the dictionary of links should be rendered."""
+        if self._when_func:
+            return bool(self._when_func(obj, context))
+        else:
+            return True
+
+    def expand(self, obj, context):
+        """Expand the thumbs size dictionary of URIs."""
+        vars = {}
+        vars.update(deepcopy(context))
+        record = obj
+        pid_value = record.pid.pid_value
+        file_key = get_record_thumbnail_file(record=record)
+        vars.update(
+            {
+                "uuid": f"record:{pid_value}:{file_key}",
+                "region": "full",
+                "rotation": "0",
+                "quality": "default",
+                "image_format": "jpg",
+            }
+        )
+        links = {}
+        for size in self._sizes:
+            vars["size"] = f"^{size},"  # IIIF syntax
+            links[str(size)] = self.link_for_thumbnail.expand(record, vars)
+        return links
+
+
+record_doi_link = ConditionalLink(
+    cond=is_datacite_test,
+    if_=RecordPIDLink("https://handle.stage.datacite.org/{+pid_doi}", when=has_doi),
+    else_=RecordPIDLink("https://doi.org/{+pid_doi}", when=has_doi),
+)
+
+
+def vars_preview_html(drafcord, vars):
```

Comment:
`drafcord` denotes a `draft` or a `record` . It is perhaps too "cute", but using `record` to denote a `draft` or a (published) `record` because both are records is too confusing.

Reply:
sounds like the plushie version of graf dracula :vampire: